js/app.js

```diff
@@ -0,0 +1,189 @@
+        this.insertImage()// inserto imagen coche
+        this.setListeners()
+        this.start()
+    },
+    setListeners() {
+        document.onkeyup = e => {
+            e.key === 'ArrowLeft' ? this.car.moveLeft() : null
+            console.log(e.key)
+            e.key === 'ArrowRight' ? this.car.moveRight() : null
+        }
+    },
+
+    start() {
+        this.car = new Car(this.ctx, this.canvasSize.w / 2 - 40, this.canvasSize.h - 100, 80, 100, 2, this.canvasSize)
+        setInterval(() => {
+            //this.clearScreen() // Las comento porque me da error. no doy con el !!!
+            //this.moveAll()
+            this.drawAll()
+
+            // this.framesCounter++
+            // this.framesCounter % 20 === 0 ? console.log('OBSTÁCULO VAAAA') : null
+        }, 1000 / 50)
+
+    },
+    setContext() {
+        this.canvasDOM = document.querySelector("#canvas") //establezco con el id = canvas el tamaño del DOM
+        this.ctx = this.canvasDOM.getContext('2d') // LINEA OBLIGATORIA para que sepa que es en 2d
+    },
+    setDimensions() {
+        this.canvasSize.w = 500 // esta ya definida arriba
+        this.canvasSize.h = 700 // esta ya definida arriba
+        this.canvasDOM.setAttribute('width', this.canvasSize.w) //establezco ancho y alto
+        this.canvasDOM.setAttribute('height', this.canvasSize.h) // LINEA OBLIGATORIA JUNTO CON LA DE ARRIBA
+    },
+    drawRoad() {
+        //console.log("funciono")
+        //console.log(this.canvasSize)
+        //void ctx.fillRect(x, y, width, height);
+        this.ctx.fillStyle = "green"// realizo un rectangulo verde que ocupe todo el canvasSize
+        this.ctx.fillRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+        this.ctx.fillStyle = "grey"
+        this.ctx.fillRect(30, 20, this.canvasSize.w - 60, this.canvasSize.h - 50)
+
+        //LINEAS BLANCAS LATERALES
+        this.ctx.strokeStyle = 'white'
+        this.ctx.lineWidth = 6
+        this.ctx.beginPath() // Esta función sirve para decirle al contexto del canvas que vamos a empezar a dibujar un camino
+        this.ctx.moveTo(10, 0) //this.ctx.moveTo(posicion x, posicion y) INICIAL DE LA LINEA
+        //this.ctx.lineTo(posicion x, posicion y) FINAL DE LA LINEA DE LA LINEA
+        this.ctx.lineTo(10, 700)
+        this.ctx.stroke() // DIBUJA EL TRAZO DEL PUNTO INICIAL AL FINAL
+        //this.ctx.closePath() //
+
+        this.ctx.strokeStyle = 'white'
+        this.ctx.lineWidth = 6
+        this.ctx.beginPath()
+        this.ctx.moveTo(490, 0) // posicion inicial x, y
+        this.ctx.lineTo(490, 700) // posicion final x, y
+        this.ctx.stroke()
+        //this.ctx.closePath() //NO necesaria
+
+        //LINEA CENTRAL DISCONTINUA
+
+        this.ctx.lineWidth = 10
+        this.ctx.beginPath()
+        this.ctx.setLineDash([60, 30])
+        this.ctx.moveTo(this.canvasSize.w / 2, 20)
+        this.ctx.lineTo(this.canvasSize.w / 2, this.canvasSize.h - 30)
+        this.ctx.stroke()

```

Comment:
El discontinuo de todas las lineas es porque necesitas resetear la forma de pintarlas. Porque si no a la siguiente vuelta del intervalo sigue pensando que ha de pintar en discontinuo.

```suggestion
        this.ctx.setLineDash([60, 30])
        this.ctx.moveTo(this.canvasSize.w / 2, 20)
        this.ctx.lineTo(this.canvasSize.w / 2, this.canvasSize.h - 30)
        this.ctx.stroke()
        this.ctx.setLineDash([0, 0])
```